### ISSUE TITLE
build error: Cannot install into target '/usr/local/var/singularity/mnt/container' mounted with noexec or nodev

### ISSUE THREAD

**Issue by cokelaer** [singularityhub/singularityhub.github.io#71]
## Link to Container Collection Log, Build, or Collection (in that order)

https://www.singularity-hub.org/collections/105

### Behavior when Building Locally

Container should be built without error

### Error on Singularity Hub

https://www.singularity-hub.org/containers/464/log

E: Cannot install into target '/usr/local/var/singularity/mnt/container' mounted with noexec or nodev

### What do you think is going on?

Works locally but on the singularityhub there seem to be a permission error?

**Comment by vsoch** [comment]
hey @cokelaer I think it comes down to something to do with [this line](https://github.com/singularityware/singularity/blob/3e338d75003f2a0a621c352bb4da2f0fc2ad9e50/libexec/bootstrap-scripts/deffile-driver-debootstrap.sh#L84) within the context of our secure build environment. @cclerget could this be the case?

As a workaround, you could try bootstrapping a docker image created from the same OS.

**Comment by cokelaer** [comment]
seems to work locally, I will push and let you know the outcome.

**Comment by vsoch** [comment]
okay great! I think @cclerget knows about the issue and has been working on it, so we will get to fixing it up when that's done!

**Comment by cokelaer** [comment]
it seems I having trouble with the new system in rebuilding the container. 
I pushed a new empty commits a few times, switch to "commit" or "manual" mode but this does not seem to restart the build.  Maybe there is some latencies ? (https://www.singularity-hub.org/collections/105)

**Comment by vsoch** [comment]
The commits only will build if there are changes to the Singularity files. It should work if you switch to manual and then select a particular commit, however. Would you mind trying again? And tell me what you did? I'll watch the server.

**Comment by vsoch** [comment]
ok I can confirm seeing you went to the collection:

```
uwsgi_1   | [pid: 64|app: 0|req: 3375/3375] 96.10.76.78 () {46 vars in 951 bytes} [Fri Oct 20 12:00:14 2017] GET /collections/105 => generated 24098 bytes in 63 msecs (HTTP/1.1 200) 5 headers in 279 bytes (1 switches on core 0)
```
but then did you push? This is the github hook trigger, and your collection is set to build manual.
```
uwsgi_1   | [pid: 64|app: 0|req: 3376/3376] 192.30.252.45 () {42 vars in 632 bytes} [Fri Oct 20 12:00:18 2017] POST /hooks/build/push => generated 63 bytes in 456 msecs (HTTP/1.1 200) 3 headers in 100 bytes (1 switches on core 0)
```

**Comment by vsoch** [comment]
hey @cokelaer I just updated the builder - would you change the builder in your settings to:

![image](https://user-images.githubusercontent.com/814322/31842620-dfd632e8-b5a3-11e7-9eb6-015c05d907ca.png)

and then try again? Note that this new builder (and changing the builder generally) hasn't been thoroughly tested so thanks in advance for your patience! :)

@katakombi this should be tested with your dolmades repo as well!

**Comment by cokelaer** [comment]
I did not manage to trigger the build again after many trials. So, I decided to delete the collection and start again from scratch. After creating the collection again, and added the master and develop branches. Push a commit. However, the build do not appear anymore. This is a different issue. I can close and open a new one if you want. The new collection is at https://www.singularity-hub.org/collections/114

**Comment by vsoch** [comment]
hey @cokelaer I've been actively (as we speak) seeing the errors that you are triggering and fixing them! If you are able to try again, it would be fantastic. The builds weren't triggering because there was a bug in the function to parse a hook.

**Comment by cokelaer** [comment]
it has started. thanks. It may take a while. I'll let you know the outcome tomorrow ;-) !

**Comment by vsoch** [comment]
Awesome! And apologies for the burps and hiccups - when I see a server error I usually respond right away looking into it, and that means a few server restarts with fixes. The change I had added was to prevent old recipes that were added and later removed from still building. The error was forgetting to check if we already had seen it, and then trying to remove a key that didn't exist in a dictionary. Derp :P  Thanks for your help and patience! We will get it working smoothly :)
